Favor project-specific build scripts over top-level script (#29918)
dotnet/aspnetcore: docs/BuildFromSource.md

When developing in VS Code, you'll need to use the `build.cmd` or `build.sh` scripts in order to build the project. You can learn more about the command line options available, check out [the section below](using-dotnet-on-command-line-in-this-repo).

> After navigating to the parent directory of the desired project, you can run the build script for that individual project.

Reviewer: Does "parent directory" here mean something like "closest ancestor directory that contains a `build.cmd`/`build.sh` file"? Many of the `.csproj` files don't have a build script in their immediate parent directory.

Author: Correct -- I've added some more text/samples here to explain the relationship between the build scripts and the project modified.

### Running tests on command-line

> Tests are not run by default. Within an project's parent directory, use the `-test` option to run tests in addition to building.

Reviewer: TBH I find it a bit confusing phrased as-is, because it's not saying what you're using `-test` with.

```suggestion
Tests are not run by default. When invoking a `build.cmd`/`build.sh` script, use the `-test` option to run tests in addition to building.
```
